File: client_app/lib/models/dish.dart
import 'package:client_app/models/dish_type.dart';
import 'package:flutter/material.dart';

class Dish{
  int id;
  String title;
  String description;
  int quantity;
  int price;
  int cookTime;
  DishType dishType;

  Dish({
    @required this.id,
    @required this.title,
    @required this.description,
    @required this.quantity,
    @required this.price,
    @required this.cookTime,
    @required this.dishType,
  });

  static Dish fromMap(Map<String, dynamic> dishMap){
    return Dish(
      id: dishMap['id'],
      title: dishMap['title'],
      description: dishMap['description'],
      quantity: dishMap['quantity'],
      price: dishMap['price'],
      cookTime: dishMap['cookTime'],
      dishType: DishType.fromMap(dishMap['dishType'])
    );
  }
}

File: order-mgmt/src/main/kotlin/com/order/ordermgmt/models/Dish.kt
package com.order.ordermgmt.models

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import javax.persistence.*

@Entity
class Dish(
        val title:String,
        val description: String,
        val quantity:Int,
        val price: Int,
        val cookTime: Int,
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name="dish_type_id")
        val dishType:DishType,
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "restaurant_id")
        val restaurant:Restaurant
    ):BaseEntity(){
    @ManyToMany(mappedBy = "dishList")
    @JsonIgnore
    val orderList= mutableListOf<ClientOrder>()
}